lib/jazzy/doc_builder.rb

```diff
@@ -156,6 +173,21 @@ def self.decl_for_token(token)
       end
     end
 
+    def self.puts_lint(undocumented)
+      tokens_by_file = undocumented.group_by do |d|
+        if d['key.filepath']
+          Pathname.new(d['key.filepath']).basename.to_s
+        else
+          d['key.modulename'] || ''
+        end
+      end
+      tokens_by_file.each_key do |file|
+        tokens_by_file[file].each do |token|
+          puts token['key.filepath'] + "\t" + token['key.doc.line'].to_s + "\t" + token['key.kind'] + "\t" + token['key.name']
```

Comment:
The advantage of %w is that it uses whitespace to separate the strings in the array, so you don’t need to quote each one.

It’s very much a matter of taste in the Ruby community. I came to prefer it after some initial reluctance.

> On Nov 22, 2015, at 5:34 PM, Jeff Verkoeyen <notifications@github.com> wrote:
> 
> In lib/jazzy/doc_builder.rb <https://github.com/realm/jazzy/pull/390#discussion_r45566878>:
> 
> > @@ -156,6 +173,21 @@ def self.decl_for_token(token)
> >        end
> >      end
> >  
> > +    def self.puts_lint(undocumented)
> > +      tokens_by_file = undocumented.group_by do |d|
> > +        if d['key.filepath']
> > +          Pathname.new(d['key.filepath']).basename.to_s
> > +        else
> > +          d['key.modulename'] || ''
> > +        end
> > +      end
> > +      tokens_by_file.each_key do |file|
> > +        tokens_by_file[file].each do |token|
> > +          puts token['key.filepath'] + "\t" + token['key.doc.line'].to_s + "\t" + token['key.kind'] + "\t" + token['key.name']
> Is there an advantage of %w(...) over [...]?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/realm/jazzy/pull/390/files#r45566878>.
>